php-classes/Slate/CBL/Skill.php

```diff
@@ -111,4 +122,24 @@ public function save($deep = true)
             $this->Competency->getTotalDemonstrationsRequired(true); // true to force refresh of cached value
         }
     }
+    
+    public function getCompetencyDescriptor()
+    {
+        return $this->Competency ? $this->Competency->Descriptor : null;
+    }
+    
+    public function getCompetencyCode()
+    {
+        return $this->Competency ? $this->Competency->Code : null;
+    }
+    
+    public function getCompetencyLevel()
+    {
+        $level = null;
+        if ($GLOBALS['Session']->PersonID && $this->Competency && $StudentCompetency = StudentCompetency::getByWhere(['StudentID' => $GLOBALS['Session']->PersonID, 'CompetencyID' => $this->Competency->ID])) {
```

Comment:
@nbey this is going to have really poor performance characteristics at scale. I've learned that instead of leaning on dynamicfields too much -- which are easy to implement but super naive because there's zero coordination between populating them for every record in a result -- it's often better to create a UI-centric endpoint that can use optimized joins to deliver a specialized result set efficiently.

You can see this pattern with the demonstrations dashboards, they use the standard CRUD endpoints as much as possible for loading their auxiliary data, but for their primary datasets that are a novel combination of data from across tables there are a couple endpoints within the `Slate\CBL\Demonstartions\*DashboardRequestHandler` classes. 

I setup all the endpoints for the /cbl/dashboards/tasks/* apps as simple inline PHP scripts that just output the template, but at some point they should get *DashboardRequestHandler classes of their own like the /cbl/dashboards/demonstrations/* routes do that handle both loading the app and providing for a couple specialized data endpoints